Wikipedia/Code/TalkPagesController.swift

```diff
@@ -54,59 +48,49 @@ class TalkPageController {
     }
     
     func fetchTalkPage(completion: ((Result<TalkPage, Error>) -> Void)? = nil) {
-        
         guard let urlTitle = type.urlTitle(for: title, titleIncludesPrefix: titleIncludesPrefix),
             let taskURL = fetcher.taskURL(for: urlTitle, host: host) else {
             completion?(.failure(TalkPageError.createTaskURLFailure))
             return
         }
         
-        var localTalkPage: TalkPage?
-        localHandler.dataStore.viewContext.performAndWait {
+        let moc = dataStore.viewContext
+        moc.perform {
```

Comment:
I thought you could skip the `perform` block when accessing the main context from the main queue - is that incorrect or is the assumption that this could be called from a non-main queue? 🤔

Reply:
You're correct. The intent here was to reduce dependence on the moc being the `viewContext` but this perform could be removed